pkg/provider/aws/util/provider.go

```diff
@@ -45,3 +45,19 @@ func GetAWSProvider(store esv1beta1.GenericStore) (*esv1beta1.AWSProvider, error
 	}
 	return prov, nil
 }
+
+func IsReferentSpec(prov esv1beta1.AWSAuth) bool {
+	if prov.JWTAuth != nil && prov.JWTAuth.ServiceAccountRef != nil && prov.JWTAuth.ServiceAccountRef.Namespace == nil {
+		return true
+	}
+	if prov.SecretRef != nil &&
+		prov.SecretRef.AccessKeyID.Namespace == nil &&
+		prov.SecretRef.SecretAccessKey.Namespace == nil &&
+		(prov.SecretRef.SessionToken == nil ||
+			(prov.SecretRef.SessionToken != nil &&
+				prov.SecretRef.SessionToken.Namespace == nil)) {
+		return true
+	}
```

Comment:
What happens if `AccessKeyID` is referrent, but `SecretAccessKey` is not?

Also (this one might be me lacking AWS basics), can't SessionToken be used by itself?

Reply:
I think we don't really want to allow users to have AccessKeyID as referent but SecretAccessKey as not, so maybe we could add that in the `ValidateStore` method. WDYT?

Reply:
> can't SessionToken be used by itself

When you make a call using temporary security credentials, the call must include a session token, which is returned along with those temporary credentials. AWS uses the session token to validate the temporary security credentials.

From: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html

It must be used in addition with the other things. 


> On the other side, it seems to be more simple to just allow this mix-and-match

I think this will be easier to maintain. I don't really see why one would want to mix-and-match, though :shrug: